### ISSUE TITLE
useStrapiUser or fetchUser Reload Issue

### ISSUE THREAD

**Issue by ceodevforce** [nuxt-community/strapi-module#229]
### Version
@nuxtjs/strapi: 
nuxt: 

### Reproduction Link

```
const user = useStrapiUser()
```
On page reload the user disappears even after cookie options have been modified. Using for both hooks did not yield an better result. Can someone provide some insight to this?

**Comment by benjamincanac** [comment]
Hi @ceodevforce,

Have you tried this cookie options?

```
cookie: {
  path: '/'
}
```

**Comment by ceodevforce** [comment]
> Hi @ceodevforce,
> 
> Have you tried this cookie options?
> 
> ```
> cookie: {
>   path: '/'
> }
> ```

I have tried that and along with setting the maxAge; the JWT always removes after a page reload.

**Comment by benjamincanac** [comment]
Then, is the response of the `/users/me` http call a `200`?

**Comment by fabien-lenoble** [comment]
Hello I have the same issue as @ceodevforce. The jwt token cookie is removed after a page reload even after setting maxAge to allow it to persists after session is closed.

login.vue
`
const { login } = useStrapiAuth()
const user = useStrapiUser()

async function submit() {
      await login({
          identifier: email.value,
          password: password.value
      })
}
`

On submit /auth/local is called and returns a 200, jwt_token cookie is set and then /users/me is called and returns a 200.
If I reload the page, I can see the cookie being removed just before the refresh actually happens. I could not find why this was happening (I tried logging stuff in the places that set the token.value in the various composables of the module).

I then have to use login to retrieve and set the cookie again.

Thanks in advance for the help.

**Comment by benjamincanac** [comment]
@AryeBorgen The module automatically removes the `strapi_jwt` cookie when an error occurs on `/users/me` http call as you can see here: https://github.com/nuxt-community/strapi-module/blob/dev/src/runtime/composables/useStrapiAuth.ts#L37, you have to make sure your Strapi API returns a **200** on `/users/me`.

**Comment by benjamincanac** [comment]
@ceodevforce @fabien-lenoble Could you please provide a reproduction and more informations? What version of Nuxt are you using?

**Comment by fabien-lenoble** [comment]
@benjamincanac I took another look after reading your comment about the /users/me call failing and added a console.log at this exact line you linked to. I found out that when my call was made server side (I remember now reading that a fetchUser is automatically called at the start) it did actually generate an unrelated error (request to https://{myurl}.com/api/users/me failed, reason: unable to verify the first certificate ()). This error does not occur on client-side which is why I didn't dig in that direction. I will have to look into that, but that helped a lot actually! Thanks.

**Comment by benjamincanac** [comment]
@BayBreezy Have you tried setting the `cookie.path` to `/` as mentioned here https://github.com/nuxt-modules/strapi/issues/229#issuecomment-1048695721?

**Comment by titan-65** [comment]
@BayBreezy wah gwan! When I opened the issue I kept getting the same error. I am on a Mac as well at the time. So if it works on Windows then what is the problem and I totally forgot about this issue. I swear it was resolved!

**Comment by benjamincanac** [comment]
Would you be able to provide a reproduction? Is the issue really related to nuxt, do you see the `/users/me` call in your strapi logs?

Not related but is your populate working with this syntax? I would have put `populate: ['deep', '3']`.

**Comment by BayBreezy** [comment]
Waa gwan g!!? @titan-65 

@benjamincanac Yes, the populate works because of a plugin i use [Populate Deep](https://market.strapi.io/plugins/strapi-plugin-populate-deep).

I see the `user/me` call right after I log in. But when i refresh, no additional request is sent to strapi from nuxt. 
I open the devtool and check the cookie, it gets added right after I log in. But as soon as i reload the page, its gone.

**Comment by BayBreezy** [comment]
@benjamincanac @titan-65 The node version is the problem. I installed NVM on this mac and changed to version 16.x and strapi is working as it should. 

The user cookie stays in place even after a page refresh. So anything above Node 16.19.1 may not work as intended.
I am not sure if this is a nuxt issue. @danielroe , have you seen any weird behaviour when using nuxt with node version higher than 16.19.1?

**Comment by danielroe** [comment]
I have seen this bug report: https://github.com/nuxt/nuxt/issues/19194.

I've not yet had a chance to look, but I did also wonder if some issues reported possibly are related to using undici in node 18.

cc: @pi0

**Comment by titan-65** [comment]
> @benjamincanac @titan-65 The node version is the problem. I installed NVM on this mac and changed to version 16.x and strapi is working as it should. 
> 
> 
> 
> The user cookie stays in place even after a page refresh. So anything above Node 16.19.1 may not work as intended.
> 
> I am not sure if this is a nuxt issue. @danielroe , have you seen any weird behaviour when using nuxt with node version higher than 16.19.1?
> 
> 

Thanks man I will check it out on my machine. I am trying to remember if had a similar using something else.

**Comment by sedrickcz** [comment]
@BayBreezy @titan-65 I can confirm that too. On node v18.4 it's not working.  But if I use latest v16.19 it works just fine

**Comment by creazy231** [comment]
can confirm that it works on node v16.x but doesn't work on v18.16

**Comment by Dell-it** [comment]
It looks like this is a problem of node v18.*
https://github.com/node-fetch/node-fetch/issues/1624#issuecomment-1235826631

tldr: change "http://localhost:1337" on "http://127.0.0.1:1337"

**Comment by BayBreezy** [comment]
@Dell-it Thank you. You are correct. Doing this `tldr: change "http://localhost:1337/" -> "http://127.0.0.1:1337/"` actually fixed the issue for node 18.x. @benjamincanac , maybe we should add this to the docs? or update the default to this?

Thanks!
